flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -686,6 +686,8 @@ private void setVertexConfig(
             }
         }
         config.setInputs(inputConfigs);
+        config.setSupportsUnalignedInput(
+                inEdges.isEmpty() || inEdges.get(0).supportsUnalignedCheckpoints());
```

Comment:
why do we check only the first edge?

Reply:
Because if any input edge is pointwise, no input edge supports UC. However, ideally I'd move both changes of the generators for `supportsUnalignedCheckpoints` into the same generator, so it's easier to see.

Reply:
Changed it by using a forced aligned checkpoint. So not valid anymore.